src/bfd-disasm.cpp

```diff
@@ -16,34 +14,17 @@
 
 namespace bpftrace {
 
-BfdDisasm::BfdDisasm(std::string &path) : size_(0)
+BfdDisasm::BfdDisasm(std::string &path) : fs_{ path }, size_{ 0 }
 {
-  fd_ = open(path.c_str(), O_RDONLY);
-
-  if (fd_ >= 0) {
-    struct stat st;
-
-    if (fstat(fd_, &st) == 0)
-      size_ = st.st_size;
-  }
+  std::error_code ec;
+  size_ += std_filesystem::file_size(fs_, ec);
```

Comment:
Can we check `ec` for errors just in case? Don't see anything on cppreference on return value if there's an error.

Reply:
@danobi sure